lib/modules/manager/cnb/extract.ts

```diff
@@ -0,0 +1,96 @@
+import is from '@sindresorhus/is';
+import { logger } from '../../../logger';
+import { regEx } from '../../../util/regex';
+import { getDep } from '../dockerfile/extract';
+import type {
+  ExtractConfig,
+  PackageDependency,
+  PackageFileContent,
+} from '../types';
+import { type ProjectDescriptor, ProjectDescriptorToml } from './schema';
+
+const dockerPrefix = regEx(/^docker:\/\//);
+
+function isDockerRef(ref: string): boolean {
+  const schemaMatch = regEx(/^([a-z0-9]+):\/\//).test(ref);
+  if (schemaMatch && !ref.startsWith('docker://')) {
+    return false;
+  }
+
+  return true;
+}
+
+function parseProjectToml(content: string): ProjectDescriptor | null {
+  const res = ProjectDescriptorToml.safeParse(content);
+  if (res.success) {
+    return res.data;
+  }
+
+  logger.debug(res.error, 'Failed to parse buildpacks project descriptor TOML');
```

Comment:
```suggestion
  logger.debug({ err: res.error }, 'Failed to parse buildpacks project descriptor TOML');
```
you should pass the package filename, so it's easier to find the failing file